quickwit/quickwit-query/src/query_ast/visitor.rs

```diff
@@ -112,3 +112,123 @@ pub trait QueryAstVisitor<'a> {
         Ok(())
     }
 }
+
+/// Simple trait to implement a Visitor over the QueryAst.
+pub trait QueryAstMutVisitor {
```

Comment:
The name, in combination to the other QueryAstVisitor does not make sense.
`QueryAstVisitor` is also `&mut self`.

Maybe we call that a `QueryAstTransformer`